Honeycomb Bravo.lua

```diff
@@ -116,396 +133,617 @@ function array_has_true(array)
+	parking_brake = dataref_table('laminar/B738/annunciator/parking_brake')
+	doors = dataref_table('laminar/B738/annunciator/six_pack_doors')
+	cabin_door = dataref_table('laminar/B738/toggle_switch/flt_dk_door')
+elseif PROFILE == "FF/757" then
+	master_caution = dataref_table('inst/warning')
+	doors_array[0] = dataref_table('anim/door/FL')
+	doors_array[1] = dataref_table('anim/door/FR')
+	doors_array[2] = dataref_table('anim/door/ML')
+	doors_array[3] = dataref_table('anim/door/MR')
+	doors_array[4] = dataref_table('anim/door/BL')
+	doors_array[5] = dataref_table('anim/door/BR')
+	doors_array[6] = dataref_table('1-sim/anim/doors/cargoBack/anim')
+	doors_array[7] = dataref_table('1-sim/anim/doors/cargoSide/anim')
+	doors_array[8] = dataref_table('anim/doorC') -- 757 freighter cargo door
+	volt_low = dataref_table('sim/cockpit2/electrical/bus_volts')
+	volt_low_min = 28
+elseif PROFILE == "FF/767" then
+	master_caution = dataref_table('sim/cockpit/warnings/annunciators/master_caution')
+	doors_array[0] = dataref_table('1-sim/anim/doors/FL/anim')
+	doors_array[1] = dataref_table('1-sim/anim/doors/FR/anim')
+	doors_array[2] = dataref_table('1-sim/anim/doors/BL/anim')
+	doors_array[3] = dataref_table('1-sim/anim/doors/BR/anim')
+	doors_array[4] = dataref_table('1-sim/anim/doors/cargoFront/anim')
+	doors_array[5] = dataref_table('1-sim/anim/doors/cargoBack/anim')
+	doors_array[6] = dataref_table('1-sim/anim/doors/cargoSide/anim')
+	volt_low = dataref_table('sim/cockpit2/electrical/bus_volts')
+	volt_low_min = 25
+elseif PROFILE == "Toliss/32x" then
+	master_caution = dataref_table('AirbusFBW/MasterCaut')
+	vacuum = dataref_table('sim/cockpit/misc/vacuum')
+	hydro_low_p = dataref_table('AirbusFBW/HydSysPressArray')
+	-- aux_fuel_pump_l = dataref_table('sim/cockpit/switches/fuel_pump_l')
+	-- aux_fuel_pump_r = dataref_table('sim/cockpit/switches/fuel_pump_r')
+	parking_brake = dataref_table('AirbusFBW/ParkBrake')
+	-- this dataref indicates whether the battery is charging, not the actual voltage
+	volt_low = dataref_table('sim/cockpit2/electrical/battery_amps')
+	volt_low_min = 0
+	-- canopy = dataref_table('sim/cockpit/switches/canopy_open')
+	-- doors = dataref_table('sim/cockpit/switches/door_open')
+	cabin_door = dataref_table('AirbusFBW/PaxDoorArray')
+	doors_array[0] = dataref_table('AirbusFBW/PaxDoorArray')
+	doors_array[1] = dataref_table('AirbusFBW/CargoDoorArray')
+	doors_array[2] = dataref_table('AirbusFBW/BulkDoor')
+elseif PROFILE == 'DH8D' then
+	volt_low = dataref_table('sim/cockpit2/electrical/battery_voltage_actual_volts')
+	volt_low_min = 20
+	cabin_door = dataref_table('FJS/Q4XP/Manips/CabinMainDoor_Anim')
+	doors_array[0] = dataref_table('FJS/Q4XP/Manips/CabinDoors_Anim')
+	doors_array[1] = dataref_table('FJS/Q4XP/Manips/CargoDoor_Anim')
+end
+
+local DOOR_LAST_FLASHING = -1
+local DOOR_LAST_STATE = true
+
+function handle_led_changes()
+	if bus_voltage[0] > 0 or array_has_positives(bus_voltage) then
```

Comment:
We might not need both of these checks?... the `array_has_positives` is probably enough.